### DISCUSSION TITLE
Creating a NodeJs endpoint for a Vercel Edge Function

### DISCUSSION THREAD

**Discussion by infinitejulian0** [vercel/community#1774] [Help] [answered]
### Summary

Hello! 👋 

I am building a Chrome Extension and have the backend for the Chrome extension is a separate repository. I have this repository hosted on Vercel and everything works well.

However, I want to create a Vercel edge function for an OpenAI's API and am having trouble creating a public end point using Node.js that can use this edge function. All the examples I have see on Vercel's documentation and templates hit an end point locally but I want to create the end point on node.js using express so that the Chrome extension can access it.

Is this even possible or how would I approach it?

### Additional information

I tried implementing an edge function in my express index but I get the error `Error: The Edge Function "index.js" is referencing unsupported modules:` which contains a list of modules including express. This makes me think that maybe express and edge functions are not compatible.

**Comment by amyegan** [comment, marked as answer]
Hi Julian! [Most Node.js APIs are not available](https://vercel.com/docs/concepts/functions/edge-functions/limitations#most-node.js-apis-are-not-available) in Edge Functions right now. [Edge Runtime](https://edge-runtime.vercel.sh/packages/runtime), which is built on which is built on the [Google's V8 Runtime](https://developers.google.com/apps-script/guides/v8-runtime), does not yet have support for them.
> Most libraries which use Node.js APIs as dependencies can't be used in Edge Functions yet, but we're adding support for more Node.js APIs over time

However, you can [use Express.js in Serverless Functions](https://vercel.com/guides/using-express-with-vercel) right now for your public endpoint
